source/com.microsoft.tfs.util/tests/com/microsoft/tfs/util/email/EmailAddressParserTest.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the repository root.
+
+package com.microsoft.tfs.util.email;
+
+import com.microsoft.tfs.util.StringUtil;
+
+import junit.framework.TestCase;
+
+public class EmailAddressParserTest extends TestCase {
```

Comment:
I count 20 possible error messages and 8 tests of where the expected result is `false`.